### ISSUE TITLE
Support for Alder Lake heterogeneous core distribution

### ISSUE THREAD

**Issue by Dr-Noob** [Dr-Noob/cpufetch#126]
Alder Lake has efficiency cores + performance cores, which will not be detected correctly by current cpufetch implementation (each core group also have different caches)

**Comment by dd86k** [comment]
That should be under CPUID.1Fh because there are additional types (Module, Tile (AMX?), and Die) compared to CPUID.1Bh (just Core and SMT, which my Pentium N5000 prefers).

Wish I did have an Alder Lake to test even my own utility on.

Following this + may reply if I have anything.

**Comment by Dr-Noob** [comment]
Nice, thanks. But I dont think I will work on this until I have access to an Alder Lake.

**Comment by bangbang93** [comment]
I have an Alder Lake, and there is error reporting
```
[ERROR]: Unknown microarchitecture detected: M=0x00000007 EM=0x00000009 F=0x00000006 EF=0x00000000 S=0x00000002
Please, create a new issue with this error message, the output of 'cpufetch' and 'cpufetch --debug' on https://github.com/Dr-Noob/cpufetch/issues
[ERROR]: Found invalid process: '0'
Please, create a new issue with this error message, the output of 'cpufetch' and 'cpufetch --debug' on https://github.com/Dr-Noob/cpufetch/issues

                                                              
                                                              
                                                              
                                                              Name:              12th Gen Intel(R) Core(TM) i9-12900K
                                                              Microarchitecture: Unknown
                                                              Technology:        Unknown
                                                              Max Frequency:     5.300 GHz
                                                              Cores:             12 cores (24 threads)
                                                              AVX:               AVX,AVX2
                                                              FMA:               FMA3
                                                              L1i Size:          32KB (512KB Total)
                                                              L1d Size:          48KB (768KB Total)
                                                              L2 Size:           1.25MB (12.5MB Total)
                                                              L3 Size:           30MB
                                                              Peak Performance:  1.02 TFLOP/s
                                                              
                                                              
                                                              
                                                              
```
```
[ERROR]: Unknown microarchitecture detected: M=0x00000007 EM=0x00000009 F=0x00000006 EF=0x00000000 S=0x00000002
Please, create a new issue with this error message, the output of 'cpufetch' and 'cpufetch --debug' on https://github.com/Dr-Noob/cpufetch/issues
cpufetch v1.01 (Linux x86_64 build)
12th Gen Intel(R) Core(TM) i9-12900K
- Max standard level: 0x00000020
- Max extended level: 0x80000008
- CPUID dump: 0x00090672
```

**Comment by Dr-Noob** [comment]
Cool, but I will not work on this until I can get my hands on a computer with a 12th Gen CPU where I can test everything properly.

**Comment by dd86k** [comment]
@bangbang93 My [utility](https://github.com/dd86k/ddcpuid) allows dumping the CPUID data into a table using `-r -s 4` (up to subleaf 4) if you wish to contribute raw data to this issue.

**Comment by Dr-Noob** [comment]
So this has been implemented in [cpufetch v1.03](https://github.com/Dr-Noob/cpufetch/releases/tag/v1.03) - closing the issue.
